lib/trip.rb

```diff
@@ -1,34 +1,48 @@
-require 'csv'
-
-require_relative 'csv_record'
+require "csv"
+require "awesome_print"
+require "time"
+require_relative "csv_record"
+require_relative "../specs/spec_helper"
 
 module RideShare
   class Trip < CsvRecord
-    attr_reader :id, :passenger, :passenger_id, :start_time, :end_time, :cost, :rating
+    attr_reader :passenger, :passenger_id, :start_time, :end_time, :cost, :rating, :driver_id, :driver
 
     def initialize(id:,
-      passenger: nil, passenger_id: nil,
-      start_time:, end_time:, cost: nil, rating:)
+                   passenger: nil, passenger_id: nil,
+                   start_time:, end_time:, cost: nil, rating:, driver_id: nil, driver: nil)
       super(id)
 
       if passenger
         @passenger = passenger
         @passenger_id = passenger.id
-
       elsif passenger_id
         @passenger_id = passenger_id
-
       else
-        raise ArgumentError, 'Passenger or passenger_id is required'
+        raise ArgumentError, "Passenger or passenger_id is required"
       end
 
-      @start_time = start_time
-      @end_time = end_time
+      @start_time = Time.parse(start_time)
+      if end_time
+        @end_time = Time.parse(end_time)
+        if @end_time.to_i < @start_time.to_i
```

Comment:
You can directly compare `Time` instances.

```suggestion
        if @end_time < @start_time
```